File: demo-micronout-java/src/main/java/com/uniliva/service/PessoaService.java
package com.uniliva.service;

import com.uniliva.dto.PessoaDto;
import com.uniliva.entity.Pessoa;
import com.uniliva.repo.PessoaRepository;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.StreamSupport;

@Singleton
public class PessoaService {

    @Inject
    private PessoaRepository repo;

    public PessoaDto salvar(PessoaDto dto) {
        Pessoa pessoa = repo.save(dto.toEntity());
        return pessoa.toDto();
    }

    public List<PessoaDto> buscaTodos() {
        List<PessoaDto> pessoaDtos = StreamSupport
                .stream(repo.findAll().spliterator(), false).map(Pessoa::toDto)
                .collect(Collectors.toList());
        return pessoaDtos;
    }
}

File: demo-micronout-kotlin/src/main/kotlin/com/uniliva/service/PessoaService.kt
package com.uniliva.service

import com.uniliva.dto.PessoaDto
import com.uniliva.entity.PessoaEntity
import com.uniliva.mapper.toDto
import com.uniliva.mapper.toEntity
import com.uniliva.repository.PessoaRepository
import java.util.stream.Collectors
import java.util.stream.StreamSupport
import javax.inject.Singleton


@Singleton
class PessoaService(private val repo :PessoaRepository) {

    fun salvar(dto: PessoaDto): PessoaDto? {
        val pessoa: PessoaEntity = repo.save(dto.toEntity())
        return pessoa.toDto()
    }

    fun buscaTodos(): List<PessoaDto?>? = repo.findAll().map {it.toDto()}.toList()

}